sdk/storage/azure-storage-blob/samples/client_side_encryption_keyvault.py

```diff
@@ -0,0 +1,118 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: client_side_encryption_keyvault.py
+
+DESCRIPTION:
+    This sample contains code demonstrating how to conifgure the storage blob
+    service for client side encryption, storing and retrieving the key encryption key
+    (kek) from within Azure KeyVault. This sample requires a service principal be set
+    configured with access to KeyVault, and that the vault contains a 256-bit base64-
+    encoded secret named "symmetric-key". Additionally, a number of environment
+    variables, listed below, must be set. Since these often contain sensitive information,
+    they SHOULD NOT be replaced with hardcoded values in any code derived from this sample.
+
+USAGE: python client_side_encryption_keyvault.py
+"""
+
+import base64
+import os
+import sys
+import uuid
+
+from azure.identity import ClientSecretCredential
+
+from azure.keyvault.keys.crypto import CryptographyClient, KeyWrapAlgorithm
+from azure.keyvault.keys import KeyVaultKey, KeyType
+from azure.keyvault.secrets import SecretClient
+
+from azure.storage.blob import BlobServiceClient, BlobType
+
+# Environment variable keys which must be set to run this sample
+STORAGE_URL = 'AZURE_STORAGE_URL'
+KEYVAULT_URL = 'AZURE_KEYVAULT_URL'
+CLIENT_ID = 'AZURE_CLIENT_ID'
+CLIENT_SECRET = 'AZURE_CLIENT_SECRET'
+TENANT_ID = 'AZURE_TENANT_ID'
+
+def get_env_var(key):
+    try:
+        return os.environ[key]
+    except KeyError:
+        print('{} must be set.'.format(key))
+        sys.exit(1)
+
+def make_resource_name(prefix):
+    return '{}{}'.format(prefix, str(uuid.uuid4()).replace('-', ''))
+
+class KeyWrapper:
+    """ Class that fulfills the interface used by the storage SDK's
+        automatic client-side encyrption and decryption routines. """
+
+    def __init__(self, kek, credential):
+        self.algorithm = KeyWrapAlgorithm.aes_256
+        self.kek = kek
+        self.kid = kek.id
+        self.client = CryptographyClient(kek, credential)
+
+    def wrap_key(self, key):
+        wrapped = self.client.wrap_key(key=key, algorithm=self.algorithm)
+        return wrapped.encrypted_key
+
+    def unwrap_key(self, key, _):
+        unwrapped = self.client.unwrap_key(encrypted_key=key, algorithm=self.algorithm)
```

Comment:
@bryevdv made a good point here. If the storage client expects `KeyWrapper` to try arbitrary algorithms and raise when they fail, we should play along even if I think it's a footgun 😁